### PR TITLE
Feat/login

### PR DESCRIPTION
**Pull request by Ramotka**

### CODE REVIEW

File: projects/user-auth/src/lib/adapters/primary/components/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  ) {}
+
+  onLoginSubmited(login: FormGroup): void {
+    this._addsCredentialsDto
+      .add({
+        email: login.get('email')?.value,
+        password: login.get('licenceKey')?.value,
+      })
+      .pipe(
+        tap((data) => localStorage.setItem('accessToken', data.accessToken)),
```
Comment (gradzio) [inline projects/user-auth/src/lib/adapters/primary/components/login.component.ts:52]: Local storage should be part of the storage used behind the context port

File: projects/user-auth/src/lib/adapters/primary/components/login.component.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    </mat-form-field>
+
+    <mat-form-field class="w-96">
+      <mat-label>Licence Key</mat-label>
+      <input
+        matInput
+        [type]="hide ? 'password' : 'text'"
+        formControlName="licenceKey"
+      />
+      <mat-icon matSuffix (click)="hide = !hide">{{
```
Comment (gradzio) [inline projects/user-auth/src/lib/adapters/primary/components/login.component.html:21]: I personally don't like template vars. Please make It part of the component. You can use behavior subject

**Review by gradzio — CHANGES_REQUESTED** [review]

File: projects/user-auth/src/lib/adapters/primary/components/login.component.html (hunk 2 above)
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/primary/components/login.component.html:21]: Logic moved to component. For now a simpler version without the behavior subject :)

File: projects/user-auth/src/lib/adapters/primary/resolvers/current-user-id.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  Resolve,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { Observable, of } from 'rxjs';
+
+@Injectable()
+export class CurrentUserIdResolver implements Resolve<boolean> {
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/primary/resolvers/current-user-id.resolver.ts:10]: No :)

File: projects/user-auth/src/lib/adapters/primary/guards/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { Observable, of } from 'rxjs';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/primary/guards/auth.guard.ts:11]: fixed in the next commit

File: projects/user-auth/src/lib/adapters/secondary/services/http-auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { CredentialsDTO } from '../../../application/ports/secondary/dto/credentials.dto';
+import { ResponseDTO } from '../../../application/ports/secondary/dto/response.dto';
+
+@Injectable()
+export class HttpAuthService implements AddsCredentialsDtoPort {
+  constructor(private _client: HttpClient) {}
+
+  add(credentials: CredentialsDTO): Observable<ResponseDTO> {
+    return this._client
+      .post<any>(
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/secondary/services/http-auth.service.ts:15]: Fixed in the next commit

File: projects/user-auth/src/lib/application/ports/secondary/dto/response.dto.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export interface ResponseDTO {
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply projects/user-auth/src/lib/application/ports/secondary/dto/response.dto.ts:1]: Fixed in the next commit

File: src/app/pages/home.page-module.ts [hunk 7]
```diff
@@ -21,22 +27,34 @@ import { NewFeaturePage } from './new-feature.page';
     RouterModule.forChild([
       {
         path: '',
+        canActivate: [AuthGuard],
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply src/app/pages/home.page-module.ts:30]: Fixed in the next commit

File: projects/user-auth/src/lib/adapters/primary/guards/auth.guard.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(private router: Router) {}
+
+  canActivate(
+    activatedRoute: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> {
+    if (localStorage.length === 0) {
```
[the remark this answers is not in the document]
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/primary/guards/auth.guard.ts:18]: I'm not sure how to do this

File: projects/user-auth/src/lib/adapters/primary/components/login.component.ts (hunk 1 above)
Comment (Ramotka) [reply projects/user-auth/src/lib/adapters/primary/components/login.component.ts:52]: I'm not sure how to do this

File: projects/user-auth/src/lib/adapters/primary/components/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        switchMap((response) =>
+          this._setsStateCurrentUserContext.setState({ id: response.id })
+        )
+      )
+      .subscribe(() => {
+        this._router.navigate(['/']);
+      });
+  }
+
+  hideVisibilityIcon: boolean = true;
```
Comment (gradzio) [inline projects/user-auth/src/lib/adapters/primary/components/login.component.ts:78]: We should have only observable in smart components. Please use behavior subject instead

File: projects/user-auth/src/lib/application/ports/secondary/context/selects-local-storage.context-port.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+export const SELECTS_LOCAL_STORAGE_CONTEXT =
+  new InjectionToken<SelectsLocalStorageContextPort>(
+    'SELECTS_LOCAL_STORAGE_CONTEXT'
+  );
+
+export interface SelectsLocalStorageContextPort {
+  isLoggedIn(): Observable<boolean>;
```
Comment (gradzio) [inline projects/user-auth/src/lib/application/ports/secondary/context/selects-local-storage.context-port.ts:10]: This should be select

File: projects/user-auth/src/lib/application/ports/secondary/context/selects-local-storage.context-port.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+export const SELECTS_LOCAL_STORAGE_CONTEXT =
```
Comment (gradzio) [inline projects/user-auth/src/lib/application/ports/secondary/context/selects-local-storage.context-port.ts:4]: You should not have local storage in port name. It is abstracting away from infra detail. Just call it token storage

File: projects/user-auth/src/lib/application/ports/secondary/context/sets-state-local-storage.context-port.ts [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+export const SETS_STATE_LOCAL_STORAGE_CONTEXT =
```
Comment (gradzio) [inline projects/user-auth/src/lib/application/ports/secondary/context/sets-state-local-storage.context-port.ts:4]: Same here sets state token context

File: projects/user-auth/src/lib/application/ports/secondary/context/sets-state-local-storage.context-port.ts [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+  new InjectionToken<SetsStateLocalStorageContextPort>(
+    'SETS_STATE_LOCAL_STORAGE_CONTEXT'
+  );
+
+export interface SetsStateLocalStorageContextPort {
+  setLocalStorage(key: string, value: string): void;
```
Comment (gradzio) [inline projects/user-auth/src/lib/application/ports/secondary/context/sets-state-local-storage.context-port.ts:10]: SetState

**Review by gradzio — CHANGES_REQUESTED** [review]

**Review by gradzio — APPROVED** [review]

**Pull request merged**